frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/factorial-experiment-design.component.ts

```diff
@@ -0,0 +1,603 @@
+  OnInit,
+  ChangeDetectionStrategy,
+  Output,
+  EventEmitter,
+  Input,
+  ViewChild,
+  ElementRef,
+  OnChanges,
+  OnDestroy,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators, FormArray, AbstractControl } from '@angular/forms';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import {
+  NewExperimentDialogEvents,
+  NewExperimentDialogData,
+  NewExperimentPaths,
+  ExperimentVM,
+  ExperimentCondition,
+  ExperimentPartition,
+  IContextMetaData,
+  EXPERIMENT_STATE,
+  ExperimentFactor,
+  ExperimentLevel,
+} from '../../../../../core/experiments/store/experiments.model';
+import { ExperimentService } from '../../../../../core/experiments/experiments.service';
+import { TranslateService } from '@ngx-translate/core';
+import { v4 as uuidv4 } from 'uuid';
+import { map, startWith } from 'rxjs/operators';
+import { DialogService } from '../../../../../shared/services/dialog.service';
+import { ExperimentDesignStepperService } from '../../../../../core/experiment-design-stepper/experiment-design-stepper.service';
+import {
+  ExperimentConditionAliasRequestObject,
+  ExperimentFactorFormData,
+  ExperimentFactorialDesignData,
+  ExperimentLevelFormData,
+  FactorialConditionTableRowData,
+} from '../../../../../core/experiment-design-stepper/store/experiment-design-stepper.model';
+
+@Component({
+  selector: 'home-factorial-experiment-design',
+  templateUrl: './factorial-experiment-design.component.html',
+  styleUrls: ['./factorial-experiment-design.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FactorialExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() experimentInfo: ExperimentVM;
+  @Input() currentContext: string;
+  @Input() isContextChanged: boolean;
+  @Input() animationCompleteStepperIndex: number;
+  @Output() emitExperimentDialogEvent = new EventEmitter<NewExperimentDialogData>();
+
+  @ViewChild('stepContainer', { read: ElementRef }) stepContainer: ElementRef;
+  @ViewChild('factorTable', { read: ElementRef }) factorTable: ElementRef;
+  @ViewChild('levelTable', { read: ElementRef }) levelTable: ElementRef;
+
+  subscriptionHandler: Subscription;
+
+  factorialExperimentDesignForm: FormGroup;
+  factorDataSource = new BehaviorSubject<AbstractControl[]>([]);
+  allFactors = [];
+
+  // Factor Errors
+  factorPointErrors = [];
+  factorErrorMessages = [];
+  factorCountError: string = null;
+
+  // Level Errors
+  levelPointErrors = [];
+  levelCountError: string = null;
+
+  expandedId: number = null;
+  levelIds: string[] = [];
+
+  factorDisplayedColumns = ['expandIcon', 'factor', 'site', 'target', 'removeFactor'];
+  levelDisplayedColumns = ['level', 'alias', 'removeLevel'];
+
+  // Used for condition code, experiment point and ids auto complete dropdownfilteredExpFactors$: Observable<string[]>[] = [];
+  filteredExpFactors$: Observable<string[]>[] = [];
+  filteredExpPoints$: Observable<string[]>[] = [];
```

Comment:
`ExpPoint` should be changed to `Site` or `DecisionPointSite`
`ExpId` should be changed to `Target` or `DecisionPointTarget`
`Partition` should be changed to `DecisionPoint`

There are some places where these old terms are deeply embedded and are hard to change, but when we're just creating our own local var names, we should stick to the correct terms.

Reply:
done

Reply:
done for existingPartition